streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -1641,6 +1641,17 @@ class ZStream[-R, +E, +A] private[stream] (private[stream] val structure: ZStrea
   final def mapMParUnordered[R1 <: R, E1 >: E, B](n: Int)(f: A => ZIO[R1, E1, B]): ZStream[R1, E1, B] =
     self.flatMapPar[R1, E1, B](n)(a => ZStream.fromEffect(f(a)))
 
+  /**
+   * Maps over elements of the stream with the specified effectful function,
+   * partitioned by `p` executing invocations of `f` concurrently on up
+   * to `n` partitions. Transformed elements may be reordered but the order within
+   * a partition is maintained.
+   */
+  final def mapMPartitioned[R1 <: R, E1 >: E, B, K](
```

Comment:
this signature would be nicer,  I think:
```scala
def mapMPartitioned(keyBy: A => K, buffer: Int = 16)(f: A => ZIO[...])
```

Reply:
Sure, have updated it